### PR TITLE
feat: finish off depth interval-related work

### PR DESCRIPTION
**Pull request by shrouxm**
## Description
Incorporates the changes in https://github.com/techmatters/terraso-client-shared/pull/329 to fix up all the remaining work described in https://github.com/techmatters/terraso-mobile-client/issues/502

### Checklist
- [x] Corresponding issue has been opened

### Related Issues
Should resolve, with QA, https://github.com/techmatters/terraso-mobile-client/issues/502
Needs https://github.com/techmatters/terraso-client-shared/pull/329

### CODE REVIEW

File: dev-client/src/translations/en.json [hunk 1]
```diff
@@ -473,10 +473,16 @@
...
+      "delete_modal": {
+        "title": "Delete depth?",
+        "body": "Data that has been entered at this depth will be deleted. This action cannot be undone.",
+        "action": "Delete"
+      },
       "error": {
         "overlaps": "Overlaps existing",
         "ordering": "Interval bounds error"
       },
+      "bounds_unitless": "{{start}}-{{end}}",
```
Comment (paulschreiber) [inline dev-client/src/translations/en.json:485]:
```suggestion
      "bounds_unitless": "{{start}}–{{end}}",
```

File: dev-client/src/translations/en.json [hunk 2]
```diff
@@ -473,10 +473,16 @@
...
       "bounds": "{{start}}-{{end}}{{units}}",
```
Comment (paulschreiber) [inline dev-client/src/translations/en.json:486]:
```suggestion
      "bounds": "{{start}}–{{end}}{{units}}",
```

**Review by paulschreiber — APPROVED** [review]

File: dev-client/src/translations/en.json (hunk 2 above)
Comment (shrouxm) [reply dev-client/src/translations/en.json]: @paulschreiber could you explain more why the use of the minus sign unicode codepoint and not the minus-hyphen?

Comment (paulschreiber) [reply dev-client/src/translations/en.json]: I replaced the hyphen with an en dash. En dashes are used to separate ranges of values. https://practicaltypography.com/hyphens-and-dashes.html

**Pull request merged**